community/CloudflareR2.md

```diff
@@ -0,0 +1,144 @@
+
+## Uploading files with rclone
+
+1. Open a terminal and run `rclone config` to configure rclone.
+   1. Choose “n” for new remote.
+   2. Choose a name for the remote, e.g. `R2`.
+   3. Choose `Amazon S3 Compliant Storage Providers` as the storage provider. (As of the time of writing, option `4`.)
+   4. Choose `Cloudflare R2 Storage` as the provider. (As of the time of writing, option `6`.)
+   5. Choose `Enter AWS credentials in the next step.`
+   6. Copy the `Access Key ID` and `Secret Access Key` from the R2 API Token page to the terminal.
+   7. Choose the only possible region.
+   8. Copy the endpoint from the R2 API token page to the terminal.
+   9. Do not edit the advanced config.
+   10. Check if the configuration is correct & confirm.
+
+2. Navigate in the terminal to the directory where your BlueMap *web* files are located. You should be in the directory that contains the file `index.html`.
+
+3. Run the following command to upload the files to your R2 bucket:
+   ```bash
+   rclone sync . R2:drehmal-bluemap/ --transfers=60 --checkers=120 -P
+   ```
+   - `rclone` — the program you are calling
+   - `sync` — the command to synchronize files — files which are not present in the destination will be copied there and files which are not present in the source will be deleted in the destination
+   - `.` — the current directory
+   - `R2:bluemap/` — the remote you configured in the rclone configuration plus the name of your bucket
+   - `--transfers=60` — the number of parallel transfers — BlueMap consists of numerous small files, so a high number of transfers is beneficial
+   - `--checkers=120` — the number of parallel checks — BlueMap consists of numerous small files, so a high number of checks is beneficial. Checkers compare your local files to those remote.
+   - `-P` — shows you the progress
+
+4. While the files are uploading, you can continue with the next step.
+
+## Making necessary configurations
+
+1. Go to your domain inside the Cloudflare dashboard. Click on “Rules” and then “Transform Rules” — make sure you are in the Rewrite URL tab. Click on “Create Rule”.
+
+2. Give the Rule a name, e.g. `BlueMap: Index`.
+   1. Select `Custom filter expression`
+      1. In the first field, select `Hostname`. In the Operator field, select `equals`. In the value field, enter the domain you are using for BlueMap, e.g. `map.example.com`.
+      2. Click on `And` to add another rule.
+      3. In the first field select `URI Path` and in the Operator field select `ends with`. In the value field, enter `/`.
+   2. Further down under Path, select `Rewrite to...`.
+      1. In the first field select `Dynamic` and in the second field enter `concat(http.request.uri.path, "index.html")`.
+   3. It should look similar to this: ![Screenshot of the index rule in Cloudflare]({{site.baseurl}}/assets/r2/index_rule.png)
+   4. Save the rule. This rule allows you to open the BlueMap without the need to enter `index.html` at the end of the URL.
```

Comment:
Maybe introduce the rule beforehand instead, and explain there what it is about to do

Reply:
Changed✔